noir-projects/aztec-nr/aztec/src/event/event_interface.nr

```diff
@@ -14,50 +14,136 @@ use dep::protocol_types::{
     traits::{Serialize, ToField},
 };
 
-pub struct PrivateLogContentConstraintsEnum {
-    /// The contents of the log are entirely unconstrained, and could have any values.
+/// Specifies the configuration parameters for message delivery across three key dimensions:
+///
+/// +--------------------------------------------------------------------------------------------------------+
+/// | 1. Constrained or unconstrained encryption                                                             |
+/// | - When the encryption is constrained we have a guarantee that the resulting ciphertext is formed       |
+/// |   correctly. If we don't need this guarantee and we are fine with trusting the sender, it is           |
+/// |   recommended to use unconstrained encryption as it's cheaper constraint-wise.                         |
+/// +--------------------------------------------------------------------------------------------------------+
+///
+/// +--------------------------------------------------------------------------------------------------------+
+/// | 2. Delivery mechanism                                                                                  |
+/// | - The message ciphertext can be delivered either via the private log stream or as an offchain message. |
+/// | - The private log stream is handled by the Aztec protocol, is submitted to L1 blobs and consumes DA.   |
+/// | - The offchain message mechanism is implemented by the app and hence is not expected to consume DA     |
+/// |   (imagine an app submitting all the ciphertext to cloud storage).                                     |
+/// +--------------------------------------------------------------------------------------------------------+
+///
+/// +--------------------------------------------------------------------------------------------------------+
+/// | 3. Constrained or unconstrained tagging                                                                |
+/// | - Tagging is an indexing mechanism that enables recipients to locate private logs in the system. When  |
+/// |   tagging is unconstrained, there is not a cryptographic guarantee about it being performed correctly. |
+/// |   This may prevent recipients from successfully discovering their messages. Therefore, unconstrained   |
+/// |   tagging should only be used in scenarios where the recipient is trusted or has clear incentives to   |
+/// |   ensure proper message delivery.                                                                      |
+/// | - Tagging is not used in the offchain message mechanism.                                               |
+/// +--------------------------------------------------------------------------------------------------------+
+///
+///
+/// # Message authenticity
+/// Note that we always commit to the plaintext and the recipient is therefore always able to verify
+/// the authenticity of the message no matter the delivery mechanism.
```

Comment:
This is false - there is no plaintext commitment, and with unconstrained encryption the plaintext is arbitrary. we do have commitments to events and notes, but those are very different things